### PR TITLE
Fix huge message transfer protocol

### PR DESCRIPTION
**Pull request by wesbland**
The huge message transfer protocol didn't handle multiple outstanding messages at once. This changes that protocol to hash on something that is less specific. It also adds some tests for corner cases that were also failing.

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch4/ofi

**Comment by raffenet** [comment]
test:jenkins/ch4/ofi

**Comment by raffenet** [comment]
@wesbland I think Github webhooks might be broken at the moment. Will test ASAP.

**Comment by wesbland** [comment]
I don't remember if the failures here are common or not. We've xfailed everything that's in progress ourselves. Is this passing the expected tests?

**Comment by raffenet** [comment]
I think these are all expected. I'll proceed with the review.

File: src/mpid/ch4/netmod/ofi/ofi_types.h [hunk 1]
```diff
@@ -50,10 +50,22 @@
 #define MPIDI_OFI_CONTEXT_MASK       (((1ULL << MPIDI_OFI_CONTEXT_BITS) - 1) << (MPIDI_OFI_SOURCE_BITS + MPIDI_OFI_TAG_BITS))
 #define MPIDI_OFI_SOURCE_MASK        (((1ULL << MPIDI_OFI_SOURCE_BITS) - 1) << MPIDI_OFI_TAG_BITS)
 #define MPIDI_OFI_TAG_MASK           ((1ULL << MPIDI_OFI_TAG_BITS) - 1)
+/* This value comes from the fact that we use a uint32_t in
+ * MPIDI_OFI_send_handler to define the dest and that is the size we expect
+ * from the OFI provider for its immediate data field. */
+#define MPIDI_OFI_MAX_RANK_BITS      (MPIDI_OFI_SOURCE_BITS > 0 ? MPIDI_OFI_SOURCE_BITS : 32)
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_types.h]: What about `MPIDI_Global.settings.source_bits`? I'm trying to figure out if all these rank/source bit macros are necessary...

Comment (wesbland) [inline src/mpid/ch4/netmod/ofi/ofi_types.h]: The reason for this was because you might have 0 bits for the rank in the match bits but you have up to 32 bits in the immediate data. If we don't have this macro, we'd have to add this check everywhere else.

**Comment by raffenet** [comment]
Let me get a baseline test on this now that Jenkins is back up. test:jenkins/ch4/ofi

**Comment by raffenet** [comment]
helps if I start the slurm master... test:jenkins/ch4/ofi
